crates/matrix-sdk-ui/src/timeline/algorithms.rs

```diff
@@ -39,7 +39,13 @@ impl EventTimelineItemWithId<'_> {
     /// Create a clone of the underlying [`TimelineItem`] with the given
     /// reactions.
     pub fn with_reactions(&self, reactions: ReactionsByKeyBySender) -> Arc<TimelineItem> {
-        let event_item = self.inner.with_reactions(reactions);
+        let content = self.inner.content().with_reactions(reactions);
+
+        // Do not use `Self::with_content` which may override the latest_edit_json.
+        // TODO: it's likely incorrect?
```

Comment:
I don't understand this comment -  can you expand on it? If it's incorrect, do we need to fix it before we can commit this?

Reply:
Ah yeah, something I noticed, and fixed in [another PR](https://github.com/matrix-org/matrix-rust-sdk/pull/4611); I can remove the comment now, thanks for spotting it!